napari/__main__.py

```diff
@@ -187,5 +188,46 @@ def main():
             )
 
 
+def _run_pythonw():
+    """Execute this script again through pythonw.
+    This ensures we're using a framework build of Python on macOS.
+    """
+    import pathlib
+    import subprocess
+
+    cwd = pathlib.Path.cwd()
+    python_path = pathlib.Path(sys.exec_prefix) / 'bin' / 'pythonw'
+
+    if not python_path.exists():
+        msg = (
+            'pythonw executable not found. '
+            'Please install python.app via conda.'
+        )
+        raise RuntimeError(msg)
+
+    cmd = [python_path, '-m', 'napari']
+
+    # Append command line arguments.
+    if len(sys.argv) > 1:
+        cmd.append(*sys.argv[1:])
+
+    env = os.environ.copy()
+    env["NAPARI_RUNNING_PYTHONW"] = "True"
+
+    subprocess.run(cmd, env=env, cwd=cwd)
+    sys.exit()
+
+
+def main():
+    # Ensure we're always using a "framework build" on macOS.
+    _MACOS_CONDA = sys.platform == "darwin" and "CONDA_PREFIX" in os.environ
+    _RUNNING_PYTHONW = "NAPARI_RUNNING_PYTHONW" in os.environ
```

Comment:
or `PYTHONEXECUTABLE in os.environ` ... in fact, since we're basically requiring this to be `python.app` from conda, we could use that instead of `NAPARI_RUNNING_PYTHONW` if wanted